### ISSUE TITLE
Custom OpenIdConnectProviders is broken in 1.1.8

### ISSUE THREAD

**Issue by SirMrDexter** [Azure/static-web-apps-cli#844]
Before filing this issue, please ensure you're using the latest CLI by running `swa --version` and comparing to the latest version on [npm](https://www.npmjs.com/package/@azure/static-web-apps-cli).

**Are you accessing the CLI from the default port `:4280` ?**

- [ ] No, I am using a different port number (`--port`) and accessing the CLI from that port
- [x] Yes, I am accessing the CLI from port `:4280`

> Make sure you are accessing the URL printed in the console when running `swa start`!

**ℹ️ NOTE: Make sure to enable debug logs when running any `swa` commands using `--verbose=silly`**

**Describe the bug**
When using custom identify providers with OpenID Connect, we should be able to login using the URL
/.auth/login/<configname>
Refer to docs: https://learn.microsoft.com/en-us/azure/static-web-apps/authentication-custom?tabs=openid-connect%2Cinvitations#configure-a-custom-identity-provider
But since 1.1.8, when you login using that URL you get a 404 response.
Instead when you change the login URL to /.auth/login/customOpenIdConnectProviders, then it works. This is contrary to how the URL works in the cloud.

**To Reproduce**

1. Create a new swa project from scratch
2. Add `staticwebapp.config.json` file with below content in it.
```
{
  "auth": {
    "rolesSource": "/api/my/roles",
    "identityProviders": {
      "customOpenIdConnectProviders": {
        "aadb2c": {
          "registration": {
            "clientIdSettingName": "AADB2C_PROVIDER_CLIENT_ID",
            "clientCredential": {
              "clientSecretSettingName": "AADB2C_PROVIDER_CLIENT_SECRET"
            },
            "openIdConnectConfiguration": {
              "wellKnownOpenIdConfiguration": "https://AADB2C_PROVIDER_ISSUER_URL/.well-known/openid-configuration"
            }
          },
          "login": {
            "nameClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name",
            "scopes": [ "openid", "profile" ],
            "loginParameterNames": []
          }
        }
      }
    }
  }
}
```
3. start the app using `swa start`
4. Go to http://localhost:4280/.auth/login/aadb2c

**Expected behavior**
Should get the emulator login screen with the provider name as `aadb2c`

**Screenshots**
![image](https://github.com/Azure/static-web-apps-cli/assets/803034/d878b071-6f67-4aa2-a402-95d5c959772c)
![image](https://github.com/Azure/static-web-apps-cli/assets/803034/6c07d33a-98d4-438c-99f6-d7b9778281ae)

**Desktop (please complete the following information):**

- OS: Windows
- Version 10
- Node Version 18
- SWA Cli version: 1.1.8

**Comment by jag43** [comment]
I'm also getting this with aadb2c auth. Downgrading to 1.1.7 fixes it.

``` json
  "auth": {
    "identityProviders": {
      "customOpenIdConnectProviders": {
        "aadb2c": {
          "registration": {
            "clientIdSettingName": "AZURE_CLIENT_ID",
            "clientCredential": {
              "clientSecretSettingName": "AZURE_CLIENT_SECRET"
            },
            "openIdConnectConfiguration": {
              "wellKnownOpenIdConfiguration": "https://tenant.b2clogin.com/tenant.onmicrosoft.com/B2C_1_sign_in/v2.0/.well-known/openid-configuration"
            }
          },
          "login": {
            "nameClaimType": "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name",
            "scopes": ["openid", "profile"],
            "loginParameterNames": []
          }
        }
      }
    }
  },
```

**Comment by aoscodes** [comment]
I actually just ran in to this. This appears to be because of a difference in the expectations of the emulator's code and what the "documentation" suggests.

This issue is that [these regex](https://github.com/Azure/static-web-apps-cli/blob/e0cf62044d2465ee228bb52de369fbac11866efe/src/msha/auth/index.ts#L11) only parse for `[a-z]` and your custom provider is named `"aadb2c"` Which is what [this documentation](https://learn.microsoft.com/en-us/azure/active-directory-b2c/configure-authentication-in-azure-static-app) explicitly suggests.

if you change the name to something those `regex` can parse like `"aadbc"` or something it will work. You will need to change it everywhere in your `staticwebapp.config.json`.

ie:
```
{
  "auth": {
    "identityProviders": {
      "customOpenIdConnectProviders": {
        "aadbc": {
          "registration": {
            "clientIdSettingName": "AZURE_CLIENT_ID",
            "clientCredential": {
              "clientSecretSettingName": "AZURE_CLIENT_SECRET"
            },
            "openIdConnectConfiguration": {
              "wellKnownOpenIdConfiguration": "https://xxx.b2clogin.com/xxx.onmicrosoft.com/B2C_1_xxx/v2.0/.well-known/openid-configuration"
            }
          },
          "login": {
            "nameClaimType": "emails",
            "scopes": ["openid", "profile", "offline_access"],
            "loginParameterNames": []
          }
        }
      }
    }
  },
  "navigationFallback": {
    "rewrite": "/index.html",
    "exclude": ["/public/*"]
  },
  "routes": [
    {
      "route": "/sign-in",
      "redirect": "/.auth/login/aadbc"
    },
    {
      "route": "/sign-up",
      "redirect": "/.auth/login/aadbc"
    },
    {
      "route": "/logout",
      "redirect": "/.auth/logout?post_login_redirect_uri=sign-in"
    },
    {
      "route": "/.auth/login/github",
      "statusCode": 404
    },
    {
      "route": "/.auth/login/twitter",
      "statusCode": 404
    },
    {
      "route": "/.auth/login/aad",
      "statusCode": 404
    },
    {
      "route": "/.auth/login/google",
      "statusCode": 404
    },
    {
      "route": "/.auth/login/facebook",
      "statusCode": 404
    },
    {
      "route": "/.auth/login/apple",
      "statusCode": 404
    },
  ],
  "responseOverrides": {
    "401": {
      "statusCode": 302,
      "redirect": "/.auth/login/aadbc?post_login_redirect_uri=.referrer"
    }
  }
}

```

Suggest a fix to documentation.

**Comment by AverageCakeSlice** [comment]
@aoscodes I can confirm this as well. After the `1.1.8` update, it seems that numerical characters are no longer valid and only [a-z] works. I was able to "fix" the issue by changing the provider key in my `staticwebapp.config.json` from `auth0` to `authzero` and then navigating to `/.auth/login/authzero` instead.

Downgrading to `1.1.7` is the workaround for now I suppose. StaticWebApps still appears to work in production, just not the emulator.

**Comment by jonnekleijer** [comment]
@aoscodes for reference to others who use `aadb2c` as `customOpenIdConnectProviders` and see when developing locally after a redirect: 

```
We couldn't find that page, please check the URL and try again.
```

I believe previously b2c was part of the docs with the naming of "b2c" as a proposed way to configure the authentication.
[Example projects](https://github.com/search?q=++++++%22customOpenIdConnectProviders%22%3A+%7B+++++++++%22aadb2c%22&type=code)

**Comment by davormaricdev** [comment]
Can confirm that this still happens with 2.0.1. Custom identity providers with numerical names are not supported. Affected by [regex here](https://github.com/Azure/static-web-apps-cli/blob/e0cf62044d2465ee228bb52de369fbac11866efe/src/msha/auth/index.ts#L34).

**Comment by Skleni** [comment]
Any updates on when this is going to be fixed? It seems the changes were reverted before and replied after 2.0.2: https://github.com/Azure/static-web-apps-cli/commit/ec3894989eec0b22dea3cfcfb6972efaf111ae3b

**Comment by Timothyw0** [comment]
Hi @Skleni we are prepping for a new release soon. It was reverted for v2.0.2 release for a critical change due to a deprecation. Please stay tuned for v2.0.3 soon :)

**Comment by AverageCakeSlice** [comment]
Still not fixed as of 2.0.4 😕

**Comment by aeromac** [comment]
The [docs ](https://learn.microsoft.com/en-us/azure/active-directory-b2c/configure-authentication-in-azure-static-app) still say to use aadb2c, yet it hasn't been supported for several versions
